consensus/core/src/linearizer.rs

```diff
@@ -23,15 +24,19 @@ impl Linearizer {
         Self { dag_state }
     }
 
-    /// Collect the sub-dag from a specific leader excluding any duplicates or
+    /// Collect the sub-dag and the corresponding commit from a specific leader excluding any duplicates or
     /// blocks that have already been committed (within previous sub-dags).
-    fn collect_sub_dag(
+    fn collect_sub_dag_and_commit(
         &mut self,
         leader_block: VerifiedBlock,
-        last_commit_index: CommitIndex,
-        last_commit_timestamp_ms: BlockTimestampMs,
-        last_committed_rounds: Vec<Round>,
-    ) -> CommittedSubDag {
+    ) -> (CommittedSubDag, TrustedCommit) {
+        // Grab latest commit state from dag state
+        let dag_state = self.dag_state.read();
```

Comment:
Eventually I hope we can place where the current commit state is read from DagState, and where commits are buffered in DagState, inside the same function. It give readers of the code more confidence that commit state gets updated by the last commit. We can follow up with another PR after the score refactor. In the short term, maybe we can just add an assertion that the last commit index is current index - 1.

Reply:
> Eventually I hope we can place where the current commit state is read from DagState, and where commits are buffered in DagState, inside the same function. It give readers of the code more confidence that commit state gets updated by the last commit

Hhm I am not sure I understand the above comment. Could you please elaborate a bit more?

Reply:
This does not block the PR. But when reading last commit from DagState, it is unintuitive which exact commit is being read unless pushing commits to DagState also happened in the same function, or there is verification that the last commit from DagState indeed has index - 1.